RELEASE/scripts/autoscend/paths/wildfire.ash

```diff
@@ -2,3 +2,404 @@ boolean in_wildfire()
+	//only return true if the loop needs to be restarted. which only occurs if we adv were spent on pumping water
+	if(!in_wildfire())
+	{
+		return false;
+	}
+	if(!zone_available(place))
+	{
+		return false;
+	}
+	if(place.fire_level <= target_fire)
+	{
+		return false;		//already done
+	}
+	
+	int[6] water_offset = {0, 0, 10, 30, 60, 100};		//an array of the extra costs of hosing due to price increase.
+	int hoses_needed = place.fire_level - target_fire;
+	int water_needed = wildfire_water_cost("hose");
+	water_needed = water_offset[hoses_needed] + water_needed*hoses_needed;
+	
+	//pump water. restart loop if adv were spent
+	boolean retval = LX_wildfire_pump(water_needed);
+	
+	for(int i=0; i<5; i++)		//loop a max of 5 times. the max number of times fire can be reduced
+	{
+		if(place.fire_level > target_fire)		//we are not done
+		{
+			if(!LX_wildfire_hose_once(place))	//attempt to hose. if it fails do below
+			{
+				break;	//hosing failed so stop the loop. LX_wildfire_hose_once should print an explanation on why it failed
+			}
+		}
+		else break;		//we are done
+	}
+	return retval;	//we only return true during water pumping if adv was used
+}
+
+boolean LX_wildfire_hose(location place)
+{
+	return LX_wildfire_hose(place, 2);
+}
+
+boolean LX_wildfire_water()
+{
+	//use water in a variety of ways to reduce fire levels. putting it in pre-adv is problematic since we need to spend adventures here
+	//individual location watering first
+	
+	//for stone wool. needed at level 11 but we acquire it early using [Baa'baa'bu'ran]. no qualifiers since this is always needed
+	if(LX_wildfire_hose($location[The Hidden Temple])) return true;
+	
+	if(!inKnollSign())		//knoll sign does not need to farm components for bitchin meatcar
+	{
+		if(LX_wildfire_hose($location[The Degrassi Knoll Garage])) return true;
+	}
+	
+	if(get_property("auto_getSteelOrgan").to_boolean() &&		//we want steel margarita
+	get_property("questM10Azazel") != "finished" &&				//we do not yet have it
+	internalQuestStatus("questM10Azazel") > -1)					//can not hose these until quest is started
+	{
+		if(LX_wildfire_hose($location[The Laugh Floor])) return true;		//need [imp air]
+		if(LX_wildfire_hose($location[Infernal Rackets Backstage])) return true;		//need [bus pass]
+	}
+
+	if(my_level() > 10 && zone_available($location[The Hidden Bowling Alley]))
+	{
+		LX_wildfire_hose($location[The Hidden Bowling Alley]);		//part of level 11 quest. potentially might want to go after NC instead
+	}
+	
+	//mass watering. waters all areas of a certain type (outdoor, indoor, underground) reducing fire from 5 to 2
+	if(get_property("wildfirePumpGreased").to_boolean())		//only pump and mass water if you greased the pump
+	{
+		if(LX_wildfire_dust()) return true;
+		if(my_level() > 3)
+		{
+			if(LX_wildfire_frack()) return true;
+		}
+	}
+	
+	return false;
+}
+
+boolean LA_wildfire()
+{
+	if(!in_wildfire())
+	{
+		return false;
+	}
+	
+	wildfire_rainbarrel();			//collect rainwater from barrel daily
```

Comment:
This may belong in a new PR since you have already done so much in this one. Now that #902 is merged, we should eventually add a check to refill the extinguisher once per day in wildfire. Believe it makes sense to put in LA_wildfire()

Reply:
Great point. Should we wait until the extinguisher is down to 0, or do we ever want to refill it early? IIRC some skills require 10% and some 30% charge.

Reply:
Ah good idea. I've only ever contributed to the main branch but I'll give it a shot now!

Reply:
This is not the main branch, it is a side branch on the main repository.
usually we make branches in our own private fork/repositories (which we keep up to date) and then commit it to the master/main branch on LASS repo. here the side branch is on LASS repo. so everyone with basic contributor permissions can commit to it directly